components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/ApiArtifactApiServiceImpl.java

```diff
@@ -18,24 +18,101 @@
 
 public class ApiArtifactApiServiceImpl implements ApiArtifactApiService {
 
-    public Response apiArtifactGet(String apiName, String label, String apiId, MessageContext messageContext) {
+    private static final Log log = LogFactory.getLog(ApiArtifactApiServiceImpl.class);
+
+    private ApiMgtDAO apiMgtDAO = ApiMgtDAO.getInstance();
+    private final String SUPER_TENAT_DOMAIN = "carbon.super";
+
+    @Override
+    public Response apiArtifactGet(String apiName, String version , String tenantDomain, MessageContext messageContext) {
 
         InMemoryAPIDeployer inMemoryApiDeployer = new InMemoryAPIDeployer();
+        if (tenantDomain == null){
+            tenantDomain =SUPER_TENAT_DOMAIN;
+        }
+        String apiId = null;
+        String label = null;
+        try {
+            apiId = apiMgtDAO.getGatewayAPIId(apiName,version,tenantDomain);
+            label = apiMgtDAO.getGatewayAPILabel(apiId);
+        } catch (APIManagementException e) {
+            log.error(e);
+        }
         GatewayAPIDTO gatewayAPIDTO = inMemoryApiDeployer.getAPIArtifact(apiId, label);
-        String definition;
+        String definition = null;
         JSONObject responseObj = new JSONObject();
+
         if (gatewayAPIDTO != null) {
-            definition = gatewayAPIDTO.getApiDefinition();
+            try {
+                if (gatewayAPIDTO.getEndpointEntriesToBeAdd() != null ) {
+                    JSONArray endPointArray = new JSONArray();
+                    EndpointAdminServiceProxy endpointAdminServiceProxy = new EndpointAdminServiceProxy
+                            (gatewayAPIDTO.getTenantDomain());
+                    for (GatewayContentDTO gatewayEndpoint : gatewayAPIDTO.getEndpointEntriesToBeAdd()) {
+                        if (endpointAdminServiceProxy.isEndpointExist(gatewayEndpoint.getName())) {
+                            endPointArray.put(endpointAdminServiceProxy.getEndpoints(gatewayEndpoint.getName()));
+                        } else {
+                            log.error(gatewayEndpoint.getName() + " was not deployed in the gateway");
+                        }
+                    }
+                    responseObj.put("Endpoints", endPointArray);
+                }
+
+                if (gatewayAPIDTO.getLocalEntriesToBeAdd() != null) {
+                    JSONArray localEntryArray = new JSONArray();
+                    LocalEntryServiceProxy localEntryServiceProxy = new
+                            LocalEntryServiceProxy(gatewayAPIDTO.getTenantDomain());
+                    for (GatewayContentDTO localEntry : gatewayAPIDTO.getLocalEntriesToBeAdd()) {
+                        if (localEntryServiceProxy.isEntryExists(localEntry.getName())) {
+                            localEntryArray.put(localEntryServiceProxy.getEntry(localEntry.getName()));
+                        } else {
+                            log.error(localEntry.getName() + " was not deployed in the gateway");
```

Comment:
Better to have this detail in the response

Reply:
https://github.com/wso2/carbon-apimgt/commit/2bef8ac27b023cef0271216e1b2942bc739efacc